src/charm.py

```diff
@@ -1390,7 +1390,11 @@ def push_tls_files_to_workload(self) -> bool:
         if cert is not None:
             self._patroni.render_file(f"{PATRONI_CONF_PATH}/{TLS_CERT_FILE}", cert, 0o600)
 
-        return self.update_config()
+        try:
+            return self.update_config()
+        except Exception:
+            logger.exception("TLS files failed to push. Error in config update")
+            return False
```

Comment:
We don't really check for exceptions or return value of `update_config()` in most places, we should most probably go over and handle such cases.

Reply:
Yeah.... this is what we saw with @beliaev-maksim yesterday on a troubleshooting. Tnx for the quick fix here!